apps/web-mzima-client/src/app/core/services/config.service.ts

```diff
@@ -69,15 +69,38 @@ export class ConfigService {
       )
       .pipe(
         map((data) => {
+          //-------------------------------------------------
+          // Simulating adding supported prop from backend
+          //-------------------------------------------------
+          const supportedDataSources = (dataSource: any) => {
+            const supported = data.results.filter(
+              (result: any) =>
+                result['provider-name'] === 'africastalking' ||
+                result['provider-name'] === 'mteja' ||
+                result['provider-name'] === 'httpsms' ||
+                result['provider-name'] === 'infobip' ||
+                result['provider-name'] === 'sislog',
+            );
+            return supported.filter((test: any) => test['provider-name'] === dataSource.id)[0];
+          };
+          //-------------------------------------------------
           return dataSources
             .filter(
               (dataSource: any) =>
                 data.results.findIndex((result: any) => dataSource.id === result['provider-name']) >
                 -1,
             )
             .map((dataSource: any) => {
+              //-------------------------------------------------------------
+              // Simulating change of "retired" datasource names from backend
+              //-------------------------------------------------------------
+              if (dataSource.name === 'FrontlineSMS')
+                dataSource.name = 'FrontlineSMS (shutdown & leaving soon)';
+              if (dataSource.name === 'SMSSync') dataSource.name = 'SMSSync (No longer maintained)';
+              //-------------------------------------------------
               return {
                 ...dataSource,
+                supported: !!supportedDataSources(dataSource),
                 enabled: data.results.find(
                   (result: any) => dataSource.id === result['provider-name'],
                 ).enabled,
```

Comment:
@Ifycode Is it possible to only show this if the user has the datasource configured? And leave it blank, or a non-clickable list if not configured?